lib/floki/raw_html.ex

```diff
@@ -34,7 +34,8 @@ defmodule Floki.RawHTML do
   end
 
   defp build_raw_html([], html, _encoder), do: html
-  defp build_raw_html(string, _html, encoder) when is_binary(string), do: encoder.(string)
+  defp build_raw_html(string, _html, encoder) when is_binary(string),
+    do: html_escape_text(encoder.(string))
```

Comment:
Not sure if this is needed as I see that spec already covers this.
https://github.com/philss/floki/blob/6700c14f842765eb5d1661af18fd686e87d1ddd7/test/floki_test.exs#L293